admin/class-acc_user_importer-admin.php

```diff
@@ -356,94 +301,58 @@ private function proccess_user_data ( $users ) {
 			$userCity = $user["City"] ?? '';
 			
 			//Log the info we received for this user
-			$user_info = $userContactId;
-			$user_info = $user_info . " " . $userEmail;
-			$user_info = $user_info . " home:" . $userHomePhone;
-			$user_info = $user_info . " cell:" . $userCellPhone;
-			$user_info = $user_info . " member#:" . $userMembership;
-			$user_info = $user_info . " expiry:" . $userExpiry;
-			$this->log_dual("Received " . $user_info);
+			$userInfoString = $userContactId;
+			$userInfoString .= " " . $userEmail;
+			$userInfoString .= " home:" . $userHomePhone;
+			$userInfoString .= " cell:" . $userCellPhone;
+			$userInfoString .= " member#:" . $userMembership;
+			$userInfoString .= " expiry:" . $userExpiry;
+			$this->log_dual("Received " . $userInfoString);
+
+
+			// Skip users without membership number
+			if (!is_numeric($userMembership) || !is_numeric($userContactId)) {
+				$this->log_dual("User is missing membership number or contact ID, skipping.");
+				continue;
+			}
 
 			//Create an array for the core wordpress user information
-			$user_data = array(
+			$accUserData = [
 				'first_name'	=>	$userFirstName,
 				'last_name'		=>	$userLastName,
-				'display_name'	=>	$userContactId,
+				'display_name'	=>	$userFirstName . " " . $userLastName,
 				'user_nicename'	=>	strtolower($userFirstName . "-" . $userLastName),
 				'user_login'	=>	$userContactId,
 				'user_email'	=>	$userEmail,
 				'user_pass'		=>	null,
-			);
+			];
+
+			$accUserMetaData = [
+				'home_phone' => $userHomePhone,
+				'cell_phone' => $userCellPhone,
+				'membership' => $userMembership,
+				'expiry' => $userExpiry,
+				'city' => $userCity
+			];
 			
 			//check if ID already exist
 			$user_id = username_exists($userContactId);
-			$update_this_user = false;
```

Comment:
Here when we search for the user in the database, if we fail to find a ContactID, could we implement a second search for the email?  This would prevent breaking the Outaouais website...

Reply:
We do search for the email further down, but for a different reason, we assume the user is duplicated in the ACC database and skip the import.
That said, I just looked in the WP source code and the email must be unique (like the `user_login` field), thus it should be safe to do a lookup by both and update accordingly. I'll try to update the code.

Reply:
Also, log-in via email was apparently added to WordPress 4.5, so it is part of the core functionality and does not require a plugin.